src/main/java/ch/njol/skript/ScriptLoader.java

```diff
@@ -111,7 +108,51 @@ private ScriptLoader() {}
 	
 	@Nullable
 	public static Config currentScript = null;
-	
+
+	/**
+	 * If true, a {@link PreScriptLoadEvent} will be called
+	 * right before a script starts parsing, but after
+	 * {@link ScriptLoader#currentScript} has been set
+	 * to a non-null {@link Config}.
+	 */
+	private static boolean callPreLoadEvent;
+
+	/**
+	 * A set of all the SkriptAddons that have called
+	 * {@link ScriptLoader#setCallPreloadEvent(boolean, SkriptAddon)}
+	 * with true.
+	 */
+	private static Set<SkriptAddon> preloadListeners = new HashSet<>();
+
+	/**
+	 * Sets {@link ScriptLoader#callPreLoadEvent} to the provided boolean,
+	 * and adds/removes the provided SkriptAddon from {@link ScriptLoader#preloadListeners}
+	 * depending on the provided boolean (true adds, false removes).
+	 * @param state The new value for {@link ScriptLoader#callPreLoadEvent}
+	 * @param addon A non-null SkriptAddon
+	 */
+	public static void setCallPreloadEvent(boolean state, @NonNull SkriptAddon addon) {

```

Comment:
Subscription system? Way better than I had hoped! In functions, it just takes addon instance for "future usage" and ignores it.